pkg/provisioningv2/rke2/planner/config.go

```diff
@@ -276,21 +277,30 @@ func addToken(config map[string]interface{}, entry *planEntry, tokensSecret plan
 	}
 }
 
-func addAddresses(secrets corecontrollers.SecretCache, config map[string]interface{}, entry *planEntry) {
+func addAddresses(secrets corecontrollers.SecretCache, config map[string]interface{}, entry *planEntry) error {
 	internalIPAddress := entry.Metadata.Annotations[rke2.InternalAddressAnnotation]
 	ipAddress := entry.Metadata.Annotations[rke2.AddressAnnotation]
 	internalAddressProvided, addressProvided := internalIPAddress != "", ipAddress != ""
 
-	secret, err := secrets.Get(entry.Machine.Spec.InfrastructureRef.Namespace, rke2.MachineStateSecretName(entry.Machine.Spec.InfrastructureRef.Name))
-	if err == nil && len(secret.Data["extractedConfig"]) != 0 {
+	// If this is a provisioned node (not a custom node), then get the IP addresses from the machine driver config.
+	if entry.Machine.Spec.InfrastructureRef.APIVersion == rke2.RKEMachineAPIVersion {
```

Comment:
nit: we can probably short circuit retrieving/parsing the config JSON if `internalAddressProvided && addressProvided`

Reply:
Done